lib/langchain/vectorsearch/weaviate.rb

```diff
@@ -24,6 +24,8 @@ def initialize(url:, api_key:, index_name:, llm:)
         url: url,
         api_key: api_key
       )
+
+      # Weaviate requires the class name to be Capitalized: https://weaviate.io/developers/weaviate/configuration/schema-configuration#create-a-class
       @index_name = index_name
```

Comment:
Maybe should add some validations here? Raise if it's not capitalized.

Reply:
Basically Weaviate **will** automatically capitalize `"products"` to `"Products"` when you're creating an index but then will claim that it's not found if you search the `"products"` index later.

Reply:
Right. I'm suggesting to either automatically capitalize, or raise an error if it's not capitalized already.

Reply:
I'll go with auto-capitalize then.